server/server.go

```diff
@@ -51,6 +47,7 @@ type serveSetup struct {
 	streamInterceptors []grpc.StreamServerInterceptor
 	statsHandler       stats.Handler
 	services           []func(*grpc.Server)
+	postStartHook      func(*grpc.Server)
```

Comment:
nit: Usually hook is something sync for what we wait for result, e.g. webhooks. From the usage it is looks like observer pattern, like you just notify some listeners that even happends.

I suggest to
```suggestion
	onPostStartListeners      []func(*grpc.Server)
```

Reply:
Sure, that makes sense.